### ISSUE TITLE
Panics are making the program terminate too early.

### ISSUE THREAD

**Issue by levone1** [andrewgremlich/media_organizer#1]
log is here - https://pastebin.com/X8ejiwUa

Issue described here - https://github.com/andrewgremlich/photo-organizer/issues/1

**Comment by andrewgremlich** [comment]
Thanks! I see if I can get around to doing it.

**Comment by andrewgremlich** [comment]
Are you using the binary or the repository?

**Comment by andrewgremlich** [comment]
Based off that error stack, I'm thinking if you have dates attached to the Exif of the photo.

Could you verify that? If not, could you run the new bug release I made?

[Here is a new bug release.](https://github.com/andrewgremlich/media_organizer/releases/tag/v0.3.4)

**Comment by levone1** [comment]
> Based off that error stack, I'm thinking if you have dates attached to the Exif metadata of the photo.
> 
> Could you verify that? If not, could you run the new bug release I made?
> 
> [Here is a new bug release.](https://github.com/andrewgremlich/media_organizer/releases/tag/v0.3.4)

Seems you're right - I did a test with some random photos I took, and it worked great. My questions now...
- How do I know which photos in the folder are the onnes with the data problem, and/or, could the program be adjusted to still run anyway, but just ignore those ones?
- I wonder if copy would be better than cut, when moving the pictures to the new directory, just in case ...

I haven't tried bug release yet, but will do. Thank you

**Comment by levone1** [comment]
I used exiftool to check on data, and it seems like the data is there for all of the pictures in the target directory, e.g.
![Screenshot_20211219_001](https://user-images.githubusercontent.com/43003405/146664713-b5361dc3-a635-4339-84a9-393fd7d0386d.png)
I wonder what's the issue.

I tried the new version, and the output is
`...File not relocated: "/run/media/defaultuser/BB34-0DF6/Pictures/Sir Joe_1080x2160.png"
File not relocated: "/run/media/defaultuser/BB34-0DF6/Pictures/_20210616_215832.JPG"
File not relocated: "/run/media/defaultuser/BB34-0DF6/Pictures/_20210617_101946.JPG"
File not relocated: "/run/media/defaultuser/BB34-0DF6/Pictures/_20210617_101946_1.JPG"
File not relocated: "/run/media/defaultuser/BB34-0DF6/Pictures/annie-spratt-38yKQLL11d8-unspl
ash.jpg"...`
etc., for all files. So seems like error report is good, but not sure why the error...

**Comment by andrewgremlich** [comment]
I like your feedback.

With the bug release I did have the program still running off there wasn't a date. I changed the Rust panic lines a bit.

Copying instead of moving could be an additional feature I can write in!

**Comment by andrewgremlich** [comment]
Looking at that picture, it doesn't look like it has a creation date... that info looks like what's with the file system instead of the photo...

**Comment by andrewgremlich** [comment]
Now that new error log shows me it is with another spot. I'll take a look.

**Comment by andrewgremlich** [comment]
Okay, I've discovered one more thing to check. What are the permissions on your `Pictures/` folder? I did some testing and I found that "File not relocated" appears when there is no write permission.

**Comment by levone1** [comment]
> Okay, I've discovered one more thing to check. What are the permissions on your `Pictures/` folder? I did some testing and I found that "File not relocated" appears when there is no write permission.

Correct again (kind of) - I actually had the thought that it may have been caused by the fact that the target was SD card, but permissions were 0755, so I didn't get around to testing...
Anyway, I tested by moving all of the pictures to ~/Desktop, (0775), and it worked fine. Then I tried to chmod -R /run to 0775, (SD card is /run/media/...), and got error again. I guess it's a root ownership issue, (not that that is a solution, because I wouldn't want /run to be 0775 -R anyway, but just a test ...).

Interestingly, running as root, either with sudo or su, still is unsuccessful. Maybe something in the code?
